src/flow/src/server.rs

```diff
@@ -355,23 +355,23 @@ impl FlownodeBuilder {
                 info.sink_table_name().schema_name.clone(),
                 info.sink_table_name().table_name.clone(),
             ];
-            manager
-                .create_flow(
-                    flow_id as _,
-                    sink_table_name,
-                    info.source_table_ids(),
-                    true,
-                    info.expire_after(),
-                    Some(info.comment().clone()),
-                    info.raw_sql().clone(),
-                    info.options().clone(),
-                    Some(
-                        QueryContextBuilder::default()
-                            .current_catalog(info.catalog_name().clone())
-                            .build(),
-                    ),
-                )
-                .await?;
+            let args = CreateFlowArgs {
+                flow_id: flow_id as _,
+                sink_table_name,
+                source_table_ids: info.source_table_ids().to_vec(),
+                create_if_not_exists: true,
```

Comment:
Why hard coding these parameters? `create_if_not_exists: true` etc.

Reply:
> Why hard coding these parameters? `create_if_not_exists: true` etc.

because recover should only happen on restart the `create_if_not_exists` and `or_replace` can be arbitrary value(since flow doesn't exist)
 but for the sake of consistency and to make sure recover of flow actually happen, set both to true
 (which is also fine since checks for not allow both to be true is on metasrv and we already pass that)
 those comments and a new sqlness restart tests are also added for recover flow